frontend/src/data/saga.js

```diff
@@ -25,14 +26,40 @@ function* watchSingleRequest() {
 
 function* watchAddUser(action) {
   let { name } = action.payload;
+  yield performAsyncOperation(() => firebaseInteractor.addUser(name), "id");
+}
+
+function* watchDownloadImage(action) {
```

Comment:
Hmm i wonder if you can combine watchDownloadImage with performAsyncOperation

Reply:
`performAsyncOperation` is what actually does the async. I was thinking that each of these functions that watch will call the `performAsyncOperation` when applicable